src/main/java/org/pac4j/vertx/auth/Pac4jUser.java

```diff
@@ -82,8 +82,8 @@ public int readFromBuffer(int pos, Buffer buffer) {
         final byte[] jsonBytes = buffer.getBytes(pos, pos + jsonByteCount);
         pos += jsonByteCount;
         final String json = new String(jsonBytes, StandardCharsets.UTF_8);
-        final UserProfile userProfile = (UserProfile) DefaultJsonConverter.getInstance().decodeObject(json);
-        setUserProfile(userProfile);
+        final UserProfile userProfileLocal = (UserProfile) DefaultJsonConverter.getInstance().decodeObject(json);
```

Comment:
Don't see the benefit of renaming a local final variable userProfile to userProfileLocal. Please explain the benefit of this change.